lib/routes/account.js

```diff
@@ -647,6 +721,105 @@ module.exports = function (
       }
     },
     {
+      method: 'POST',
+      path: '/account/device',
+      config: {
+        auth: {
+          strategy: 'sessionToken'
+        },
+        validate: {
+          payload: isA.alternatives().try(
+            isA.object({
+              name: isA.string().max(255).required().allow(''),
+              type: isA.string().max(16).required(),
+              pushCallback: isA.string().uri({ scheme: 'https' }).max(255).required().allow(''),
+              pushPublicKey: isA.string().length(64).regex(HEX_STRING).required().allow('')
+            }),
+            isA.object({
+              id: isA.string().length(32).regex(HEX_STRING).required(),
+              name: isA.string().max(255).optional(),
+              type: isA.string().max(16).optional(),
+              pushCallback: isA.string().uri({ scheme: 'https' }).max(255).optional(),
+              pushPublicKey: isA.string().length(64).regex(HEX_STRING).optional()
+            })
+          )
+        },
+        response: {
+          schema: {
+            id: isA.string().length(32).regex(HEX_STRING).required(),
+            name: isA.string().max(255).optional().allow(''),
+            type: isA.string().max(16).optional(),
+            pushCallback: isA.string().uri({ scheme: 'https' }).max(255).optional().allow(''),
+            pushPublicKey: isA.string().length(64).regex(HEX_STRING).optional().allow('')
+          }
+        }
+      },
+      handler: function (request, reply) {
+        log.begin('Account.device', request)
+        var payload = request.payload
+        var sessionToken = request.auth.credentials
+        db.updateDevice(sessionToken.uid, sessionToken.tokenId, payload).then(
+          function (device) {
+            reply(butil.unbuffer(device))
+          },
+          reply
+        )
+      }
+    },
+    {
+      method: 'GET',
+      path: '/account/devices',
+      config: {
+        auth: {
+          strategy: 'sessionToken'
+        },
+        response: {
+          schema: isA.array().items(isA.object({
+            id: isA.string().length(32).regex(HEX_STRING).required(),
+            sessionToken: isA.string().length(64).required(),
```

Comment:
I want to flag this for extra sanity-checking.  We're now returning session-token ids that belong to other devices.  I *think* this is OK, because you can't do anything with just the id, you also need to know the corresponding session key.  But I may be missing something.  Thoughts, anyone?